File: 347-top-k-frequent-elements/top-k-frequent-elements.py
from collections import Counter
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d = Counter(nums)
        cnt = [[] for i in range(len(nums)+1)]

        for key in d:
            cnt[d[key]].append(key)
        
        ans = []

        for i in range(len(cnt)-1,-1,-1):
            for j in cnt[i]:
                if k>0:
                    ans.append(j)
                    k-=1
                else:
                    return(ans)
        
        return(ans)

File: 347-top-k-frequent-elements/top-k-frequent-elements.go
func topKFrequent(nums []int, k int) []int {
    l := make([][]int, len(nums))

    d := map[int]int{}
    ans := []int{}

    for _, v := range nums {
        if _, found := d[v]; found {
            d[v]++
        } else {
            d[v] = 1
        }
    }

    for key, v := range d {
        l[v-1] = append(l[v-1], key)
    }

    for i := len(l) - 1; i >= 0; i-- {
        for j := len(l[i])-1; j >= 0; j-- {
            k-=1
            ans = append(ans, l[i][j])
            if k==0 {
                return ans
            }
        }
    }

    return []int{}
}